### ISSUE TITLE
Sending bcc with isMail() creates two to:undisclosed-recipients: headers

### ISSUE THREAD

**Issue by Hk1020** [PHPMailer/PHPMailer#2128]
# Problem description

Sending mail to a bcc recipient only with $mail->isMail() creates two `To:undisclosed-recipients` headers. This causes the mail to be rejected by almost every site with 
```
550 Messages should have one or no To headers, not 2.
```
With `$mail->isSMTP()` only one `to:` header is created.

This happens with any combination of `$mail->addBCC(...)` and `$mail->addAddress(...)` that results in an empty `To` list.

# Code to reproduce

```
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

$mail = new PHPMailer;
$mail->isMail();

$mail->From = 'from@example.com';
$to = "joe@user.org";
$mail->addBCC($to, 'Joe User');

$mail->Subject = 'Here is the subject';
$mail->Body    = 'This is the message body';

$mail->preSend();
$smessage = $mail->getSentMIMEMessage();
echo $smessage;
```

# Debug output

```
Date: Wed, 2 Sep 2020 22:10:21 +0200
To: undisclosed-recipients:;
From: Root User <from@example.com>
Bcc: Joe User <joe@user.org>
Message-ID: <ezuGZ41uc1p4K6ZLYOVxFePAq7JTi5l9yq4S80obgek@orion>
X-Mailer: PHPMailer 6.1.7 (https://github.com/PHPMailer/PHPMailer)
MIME-Version: 1.0
Content-Type: text/plain; charset=iso-8859-1
To: undisclosed-recipients:;
Subject: Here is the subject

This is the message body
```

**Comment by XL-2000** [comment]
This is the same behavior as you posted here: https://github.com/PHPMailer/PHPMailer/issues/2129

**Comment by Hk1020** [comment]
> This is the same behavior as you posted here: #2129

No, it is not the same  behavior. Note the **two** `To: undisclosed-recipients:;` lines. This is a bug that keeps mail to be delivered.

**Comment by Synchro** [comment]
That does look like a bug.  I expect the reason for this is that you have to remove the `to` header when sending via `mail()`, as the `mail()` command adds one in, but the logic that does that probably gets skipped when there are no To recipients.

There is an easy workaround for now: use SMTP to localhost. This is documented as the recommended approach anyway as it's both faster and safer than using `mail()`, which is fundamentally broken.

**Comment by Hk1020** [comment]
> That does look like a bug. I expect the reason for this is that you have to remove the `to` header when sending via `mail()`, as the `mail()` command adds one in, but the logic that does that probably gets skipped when there are no To recipients.
> 

I had a brief look at the code and there are two places where the undisclosed recipients are added. Was a little convoluted so I didn’t dare to touch it. 

> There is an easy workaround for now: use SMTP to localhost. This is documented as the recommended approach anyway as it's both faster and safer than using `mail()`, which is fundamentally broken.

But it should work anyway I think.

**Comment by Synchro** [comment]
@Hk1020 Can you give this change a try please

**Comment by Hk1020** [comment]
> 
> 
> @Hk1020 Can you give this change a try please

Thanks for the fix. Seems to work fine. I just tried a few cases.
